### ISSUE TITLE
Results table - Boolean column not filterable if it contains NULLs

### ISSUE THREAD

**Issue by JoelRichardson** [intermine/intermine#641]
Sometimes a Boolean column is not filterable. It seems to have something to do with NULLs.

To see this, run the following query at beta.mousemine.org. Then try to filter the "Detected" column, e.g., for just Detected = true. The "Filter" button remains disabled. 

Now filter the results to stage TS23, which removes all the rows with Detected = NULL. The Detected column is now filterable.

One other thing I noticed: the filter control dialog is missing two buttons at the bottom (for inverting and resetting the selection).

Here's the query (only works at beta.mousemine.org):
```
<query model="genomic" view="GXDExpression.assayType GXDExpression.feature.symbol GXDExpression.feature.primaryIdentifier GXDExpression.stage GXDExpression.structure.name GXDExpression.detected GXDExpression.strength GXDExpression.genotype.symbol GXDExpression.sex GXDExpression.publication.mgiJnum" sortOrder="GXDExpression.assayType ASC" constraintLogic="A and B" >
  <constraint path="GXDExpression.feature.organism.taxonId" op="=" value="10090" code="B" />
  <constraint path="GXDExpression.feature" op="LOOKUP" value="Pax6" code="A" />
</query>
```

**Comment by alexkalderimis** [comment]
testmodel needs a case for this

**Comment by julie-sullivan** [comment]
@alexkalderimis The filter button is enabled but the checkboxes act like radio buttons, I can't select more than one. There is a case for wanting to view all FALSE and NULL values.

Also, the numbers are truncated.

![boolean](https://cloud.githubusercontent.com/assets/1663057/6527216/e69879ba-c410-11e4-903b-eaa6a4cbd2dc.png)

**Comment by alexkalderimis** [comment]
In the case of wanting to view all `false` and `null` values - the user can always exclude all `true` values!

But seriously though, I will fix this.

**Comment by alexkalderimis** [comment]
Also, this totally works for me! using:

```xml
<query model="genomic" view="Gene.secondaryIdentifier Gene.symbol Gene.primaryIdentifier Gene.organism.name Gene.proteins.isUniprotCanonical" sortOrder="Gene.secondaryIdentifier ASC" >
  <constraint path="Gene" op="LOOKUP" value="eve, zen, bib, r, h" extraValue="" code="A" />
</query>
```

**Comment by julie-sullivan** [comment]
```xml
<query name="" model="genomic" view="Protein.primaryIdentifier Protein.primaryAccession Protein.organism.name Protein.isUniprotCanonical" longDescription="" sortOrder="Protein.primaryIdentifier asc">
</query>
```

**Comment by alexkalderimis** [comment]
Another easy fix - the class at issue is the following: https://github.com/intermine/im-tables/blob/dev/src/views/facets/boolean.coffee, which is short enough to quote in full:

```coffee
PieChart = require './pie'

# The only difference between this class and a regular pie chart is the fact that
# boolean paths do not support multiple selection, which is enforced here.
module.exports = class BooleanChart extends PieChart

  initialize: ->
    super
    @listenTo @model.items, 'change:selected', @deselectOthers
    
  # Only one value can be selected at a time (unlike pie charts and histograms,
  # which model multi-selection), so if something gets selected go through all the
  # other items and deselect them.
  deselectOthers: (x, selected) -> if selected
    @model.items.each (m) -> m.deselect() if x isnt m
```

As you can see, I will simply remove this class.

**Comment by alexkalderimis** [comment]
Should be fixed in 2.0.0-beta-18

**Comment by julie-sullivan** [comment]
From @alexkalderimis 

The null value constraint issue is a problem with the intermine pathquery engine. The tables are generating the correct query:
```
<query model="genomic" view="Gene.primaryIdentifier Gene.proteins.primaryAccession Gene.proteins.isUniprotCanonical" sortOrder="Gene.primaryIdentifier ASC" >
  <constraint path="Gene.proteins.isUniprotCanonical" op="ONE OF" >
    <nullValue/>
    <value>false</value>
  </constraint>
</query>
```
But the result returned do not include any nulls. Interestingly, on string columns that include nulls, the query engine behaves correctly.
